src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java

```diff
@@ -578,6 +601,79 @@ private boolean createComponent(Channel channel, User sender, String subcomponen
+                    return null;
+                }
+
+                GitHub github = GitHub.connect();
+                GHOrganization org = github.getOrganization(IrcBotConfig.GITHUB_INFRA_ORGANIZATION);
+                GHRepository repo = org.getRepository(IrcBotConfig.GITHUB_UPLOAD_PERMISSIONS_REPO);
+
+
+                GHContent prTemplate = repo.getFileContent(".github/PULL_REQUEST_TEMPLATE.md");
+                if (prTemplate == null) {
+                    out.message("Could not retrieve PR template for " + IrcBotConfig.GITHUB_UPLOAD_PERMISSIONS_REPO);
+                    return null;
+                }
+
+                String artifactPath = getArtifactPath(channel, github, forkTo);
+                if (StringUtils.isBlank(artifactPath)) {
+                    out.message("Could not resolve artifact path for " + forkTo);
+                    return null;
+                }
+
+                if(!repo.getBranches().containsKey(branchName)) {
+                    repo.createRef("refs/heads/" + branchName, repo.getBranch("master").getSHA1());
+                }
+
+                GHContentBuilder builder = repo.createContent();
+                builder = builder.branch(branchName).message("Adding upload permissions file for " + forkTo);
+                String name = forkTo.replace("-plugin", "");
+                String content = "---\n" +
+                        "name: \"" + name + "\"\n" +
+                        "github: \"jenkinsci/" + forkTo + "\"\n" +
+                        "paths:\n" +
+                        "- \"" + artifactPath + "\"\n" +
+                        "developers:\n";
+                StringBuffer userBuffer = new StringBuffer();
+                for(String u : releaseUsers) {
+                    userBuffer.append("- \"" + u + "\"\n");
+                }
+                content += userBuffer.toString();
+
+                builder.content(content).path("permissions/plugin-" + forkTo.replace("-plugin", "") + ".yml").commit();
+
+                String prText = String.format("Hello from your friendly Jenkins Hosting Bot!%n%n" +
+                        "This is an automatically created PR for:%n%n" +
+                        "- %s/browse/%s%n" +
+                        "- https://github.com/%s/%s%n%n" +
+                        "The user listed this PR may not have logged in to Artifactory yet, check the PR status.%n" +
+                        "To check again, retrigger the build using Checks area or by closing and reopening the PR.",
+                        IrcBotConfig.JIRA_URL, jiraIssue, IrcBotConfig.GITHUB_ORGANIZATION, forkTo);
+
+                GHPullRequest pr = repo.createPullRequest("Add upload permissions for " + forkTo, branchName, repo.getDefaultBranch(), prText);
+                prUrl = pr.getHtmlUrl().toString();
+                channel.send().message("Create PR for repository updater: " + prUrl);
+            } catch (IOException e) {
```

Comment:
there's a catch 2 lines down doing the same thing
```suggestion
            } catch (Exception e) {
```